sympy/physics/control/tests/test_lti.py

```diff
@@ -0,0 +1,258 @@
+         [1 / (2 * s**2 + 5 * s + 2), (s + 1) / (s**2 + 4 * s + 4)]])
+    assert simplify(tfm.G - G) == zeros(2, 2)
+
+
+def test_StateSpaceModel_eval():
+    t, y1, y2, omega = symbols('t, Y1, Y2, omega')
+    ssm = StateSpaceModel(eye(2) * omega, zeros(2, 1),
+                          eye(2), zeros(2, 1))
+    u = eye(1) * omega
+    x0 = Matrix([y1, y2])
+    assert ssm.evaluate(u, x0, t) == Matrix([y1 * exp(omega * t),
+                                             y2 * exp(omega * t)])
+
+    ssm = ssm.subs(omega, 1)
+    u = u.subs(omega, 1)
+    x0 = x0.subs([(y1, 1.0), (y2, 0.0)])
+    sol_num = ssm.evaluate(u, x0, (t, [0, 1.0]))
+    assert sol_num == [Matrix([1.0, 0.0]), Matrix([e, 0.0])]
+
+    A = Matrix([[-1, 1], [1, -1]])
+    ssm = StateSpaceModel(A, ones(2, 1),
+                          eye(2), zeros(2, 1))
+    u = Matrix([exp(2 * t)])
+    x0 = Matrix([1, 0])
+    sol = ssm.evaluate(u, x0, t, do_integrals=True, simplify=True)
+    assert sol == Matrix([
+                         [cosh(2 * t)],
+                         [sinh(2 * t)]])
+
+    sol_num = ssm.evaluate(u, x0, (t, [0]))
+    assert sol_num == [Matrix([1.0, 0])]
+
+    sol = ssm.evaluate(u, x0, t, do_integrals=False, simplify=True)
+    tau = symbols('tau', positive=True)
+    assert sol == Matrix([
+        [Integral(exp(2 * tau), (tau, 0, t)) + 1 / 2 + exp(-2 * t) / 2],
+        [Integral(exp(2 * tau), (tau, 0, t)) + 1 / 2 - exp(-2 * t) / 2]])
+
+    sol = ssm.evaluate(u, x0, t, 1, do_integrals=False, simplify=True)
+    assert sol == Matrix([
+        [Integral(exp(2 * tau), (tau, 1, t)) + R(1, 2) + exp(-2 * t + 2) / 2],
+        [Integral(exp(2 * tau), (tau, 1, t)) + R(1, 2) - exp(-2 * t + 2) / 2]])
+
+
+def test_transferFunctionModel_eval():
+    s = symbols('s')
+    G = Matrix([[1 / s + 2 / s**2],
+                [2 * s / (s**2 + 3 * s + 5)]])
+    u = Matrix([[s + 1]])
+    assert expand(TransferFunctionModel(G).evaluate(u, s) -
+                  G * u).is_zero
+
+
+def test_StateSpaceModel_series():
+    a0, a1, a2, b0, b1, b2 = symbols('a:3, b:3')
+    ssm1 = StateSpaceModel(
+        Matrix([[0, 1],
+               [-a0, -a1]]),
+        Matrix([0, 1]),
+        Matrix([[b0, b1]]),
+        zeros(1)
+    )
+    ssm2 = StateSpaceModel(
+        Matrix([[0, 1, 0],
+                [0, 0, 1],
+                [-a0, -a1, -a2]]),
+        Matrix([0, 0, 1]),
+        Matrix([[b0, b1, b2]]),
+        zeros(1)
+    )
+    expect = StateSpaceModel(
+        Matrix([[0, 1, 0, 0, 0],
+                [-a0, -a1, 0, 0, 0],
+                [0, 0, 0, 1, 0],
+                [0, 0, 0, 0, 1],
+                [b0, b1, -a0, -a1, -a2]]),
+        Matrix([0, 1, 0, 0, 0]),
+        Matrix([[0, 0, b0, b1, b2]]),
+        zeros(1)
+    )
+    assert ssm1.series(ssm2).represent == expect.represent
+
+
+def test_TransferFunctionModel_series():
+    a, b, s = symbols('a, b, s')
+    tfm1 = TransferFunctionModel(
+        Matrix([1 / (a * s)])
+    )
```

Comment:
Isn't a more standard API to pass in a numerator and denominator polynomial?

Reply:
@moorepants, Our base class `TransferFunction` represents the transfer function model of a transfer function matrix G  in Laplace space. Right now in this PR, we have support for MIMO transfer function. Do you want support for SISO tf instead? 

But in general there are some flaws with this API:
```
>>> G = Matrix([s/(s**2 + 1),  1/(s**3 + 2)])
>>> pprint(_)
⎡  s   ⎤
⎢──────⎥
⎢ 2    ⎥
⎢s  + 1⎥
⎢      ⎥
⎢  1   ⎥
⎢──────⎥
⎢ 3    ⎥
⎣s  + 2⎦
>>> for i in G:
...     print(type(i), i.as_numer_denom())
... 
<class 'sympy.core.mul.Mul'> (s, s**2 + 1)
<class 'sympy.core.power.Pow'> (1, s**3 + 2)
>>> 
>>> for i in G:
...     print(type(i.as_numer_denom()[1]))
... 
<class 'sympy.core.add.Add'>
<class 'sympy.core.add.Add'>
```
Is it fine or should it be a `Poly`?

Reply:
i agree, we should pass in the numerator and the denominator as `Poly`. 
`G = Matrix([Poly(2*s)/Poly(2 + s**2 - s), Poly(s)/Poly(1 + s**3)])`

Reply:
> Do you want support for SISO tf instead?

We should support SISO and MIMO in some way, what way that is I'm not quite sure of until I see some possible options.

Some questions to determine answers to:

- Should numerator and denomators be stored seperately (internal to the class)? If so, why?
- Should polynomials be stored as normal SymPy expressions or Poly() objects? I think the Poly() objects are generally confusing to use for end users, but they offer more power. Maybe these need to be used internally, but not exposed to the user.
- Should we have a `sympy.Matrix()` of Transferfunction objects? Or a sympy.matrix of ratios of polynomials as you have it. Can a user create SISO transfer functions and store them in a Matrix() or do we need a MIMO transferfunction object that internally handles the Matrix? Why or why not?

Reply:
* > Should numerator and denomators be stored seperately (internal to the class)?

Please correct me if I'm wrong but I think that the common functionality won't require storing numerator and the denominator separately, it also makes it more complex. In MIMO, what would be the num and denom btw? We have a matrix of transfer functions (`Mul`). If the user would need num and denom of each rational TFs in that matrix separately, they can use `as_numer_denom` through the `Mul`.  
        
* > Should polynomials be stored as normal SymPy expressions or Poly() objects? I think the Poly() objects are generally confusing to use for end users, but they offer more power. Maybe these need to be used internally, but not exposed to the user.

I think users can pass in normal polynomial sympy expressions and we can convert them into Poly() objects internally. It will take less time for users to simply initialize a transfer function. 

* > Should we have a sympy.Matrix() of Transferfunction objects? Or a sympy.matrix of ratios of polynomials as you have it. 

I think we should have a matrix of ratios of polynomials as we have right now. Just the API would change but internally both represent the same thing, no?? 

* > Can a user create SISO transfer functions and store them in a Matrix() or do we need a MIMO transferfunction object that internally handles the Matrix?

Tbh, I'm a little confused about this one like how can we support both SISO and MIMO?

Reply:
We have almost discussed the API for `TransferFunction` class. I will first send a PR for that in this week.

Reply:
> [Poly(2*s)/Poly(2 + s**2 - s), Poly(s)/Poly(1 + s**3)])

Actually, the polynomial division like `Poly(2*s)/Poly(2 + s**2 - s)` just coerces into an `Expr`, so it can be sufficient to store them just as `Expr`. However, in some computations, you should be careful that you had expanded all terms into canonical numerator and denominator terms, because `Expr` does not automatically expand.

Reply:
And I think that it is just straightforward to consider SISO as a 1*1 matrix transfer function model from the implementation side. However, it can be added as a convenience feature to automatically put scalar expression into 1*1 matrix.
However, I don't think that it should start parsing too many optional specifications that are redundant.